File: client-app/src/app/models/orders/salesCommission.ts
export interface SalesCommission {
    salesCommissionId?: string;
    salesRequestId: string;
    saleTypeId: string;
    statusId?: string;
    commissionDate?: Date | string | null;

    projectId?: string;
    projectName?: string;
    apartmentName?: string;
    salePrice: number;
    collectedAmount?: number | null;

    salesRepPartyId?: string | null;
    salesRepName?: string | null;
    salesRepPercent: number;
    salesRepAmount?: number | null;
    salesRepNetAmount?: number | null;

    managerPartyId?: string | null;
    managerName?: string | null;
    managerPercent: number;
    managerAmount?: number | null;
    managerNetAmount?: number | null;

    salesRep2PartyId?: string | null;
    salesRep2Name?: string | null;
    salesRep2Percent?: number | null;
    salesRep2Amount?: number | null;
    salesRep2NetAmount?: number | null;

    manager2PartyId?: string | null;
    manager2Name?: string | null;
    manager2Percent?: number | null;
    manager2Amount?: number | null;
    manager2NetAmount?: number | null;

    externalCompanyPartyId?: string | null;
    externalCompanyName?: string | null;
    externalCompanyPercent?: number | null;
    externalCompanyGrossAmount?: number | null;
    externalCompanyNetAmount?: number | null;

    externalSalesRepPartyId?: string | null;
    externalSalesRepName?: string | null;
    externalSalesRepPercent?: number | null;
    externalSalesRepAmount?: number | null;
    externalSalesRepNetAmount?: number | null;
    hasExternalSalesRepWithholdingTaxExemption: boolean;
    externalSalesRepNationalId?: string | null;

    externalManagerPartyId?: string | null;
    externalManagerName?: string | null;
    externalManagerPercent?: number | null;
    externalManagerAmount?: number | null;
    externalManagerNetAmount?: number | null;
    hasExternalManagerWithholdingTaxExemption: boolean;
    externalManagerNationalId?: string | null;

    hasVatExemption: boolean;
    hasWithholdingTaxExemption: boolean;
    vatPercent: number;
    withholdingTaxPercent: number;

    notes?: string | null;
    createdStamp?: Date | string | null;
    lastUpdatedStamp?: Date | string | null;
}

export interface CommissionRateDefaults {
    salesRepPercent: number;
    managerPercent: number;
    externalCompanyPercent?: number | null;
    externalSalesRepPercent?: number | null;
    externalManagerPercent?: number | null;
}

export interface CommissionDefaults {
    projectId?: string;
    projectName?: string;
    apartmentName?: string;
    salePrice: number;
    collectedAmount: number;
    collectedRatio: number;
    existingCommissionId?: string | null;
    // Key = saleTypeId (COMM_SALE_DIRECT | COMM_SALE_PERSONAL | COMM_SALE_INDIRECT)
    rates: Record<string, CommissionRateDefaults>;
}

export interface ApprovedSalesRequestLovItem {
    salesRequestId: string;
    customerName?: string | null;
    apartmentName: string;
    projectName: string;
    totalPrice?: number | null;
}

export interface ApprovedSalesRequestEnvelope {
    items: ApprovedSalesRequestLovItem[];
    totalCount: number;
}

File: Domain/SalesCommission.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Serialization;

namespace Domain;

[JsonObject(NamingStrategyType = typeof(SnakeCaseNamingStrategy))]
public class SalesCommission
{
    public string SalesCommissionId { get; set; } = null!;

    public string SalesRequestId { get; set; } = null!;

    // COMM_SALE_DIRECT | COMM_SALE_PERSONAL | COMM_SALE_INDIRECT
    public string SaleTypeId { get; set; } = null!;

    // COMMISSION_PENDING | COMMISSION_APPROVED | COMMISSION_PAID
    public string? StatusId { get; set; }

    public DateTime? CommissionDate { get; set; }

    // -----------------------------------------------------------------
    // Sale snapshot (denormalized from SalesRequest / Product at creation)
    // -----------------------------------------------------------------
    public string? ProjectId { get; set; }
    public decimal SalePrice { get; set; }
    public decimal? CollectedAmount { get; set; }

    // -----------------------------------------------------------------
    // Internal sales rep — present for all sale types
    // -----------------------------------------------------------------
    public string? SalesRepPartyId { get; set; }
    public decimal SalesRepPercent { get; set; }
    public decimal? SalesRepAmount { get; set; }
    public decimal? SalesRepNetAmount { get; set; }

    // -----------------------------------------------------------------
    // Internal manager — present for all sale types
    // -----------------------------------------------------------------
    public string? ManagerPartyId { get; set; }
    public decimal ManagerPercent { get; set; }
    public decimal? ManagerAmount { get; set; }
    public decimal? ManagerNetAmount { get; set; }

    // -----------------------------------------------------------------
    // Second internal sales rep — DIRECT / PERSONAL split only
    // -----------------------------------------------------------------
    public string? SalesRep2PartyId { get; set; }
    public decimal? SalesRep2Percent { get; set; }
    public decimal? SalesRep2Amount { get; set; }
    public decimal? SalesRep2NetAmount { get; set; }

    // -----------------------------------------------------------------
    // Second internal manager — DIRECT / PERSONAL split only
    // -----------------------------------------------------------------
    public string? Manager2PartyId { get; set; }
    public decimal? Manager2Percent { get; set; }
    public decimal? Manager2Amount { get; set; }
    public decimal? Manager2NetAmount { get; set; }

    // -----------------------------------------------------------------
    // External broker company — INDIRECT only
    // -----------------------------------------------------------------
    public string? ExternalCompanyPartyId { get; set; }
    public decimal? ExternalCompanyPercent { get; set; }
    public decimal? ExternalCompanyGrossAmount { get; set; }
    public decimal? ExternalCompanyNetAmount { get; set; }

    // -----------------------------------------------------------------
    // External broker's sales rep — INDIRECT only
    // -----------------------------------------------------------------
    public string? ExternalSalesRepPartyId { get; set; }
    public decimal? ExternalSalesRepPercent { get; set; }
    public decimal? ExternalSalesRepAmount { get; set; }
    public decimal? ExternalSalesRepNetAmount { get; set; }
    public bool HasExternalSalesRepWithholdingTaxExemption { get; set; }
    public string? ExternalSalesRepNationalId { get; set; }

    // -----------------------------------------------------------------
    // External broker's manager — INDIRECT only
    // -----------------------------------------------------------------
    public string? ExternalManagerPartyId { get; set; }
    public decimal? ExternalManagerPercent { get; set; }
    public decimal? ExternalManagerAmount { get; set; }
    public decimal? ExternalManagerNetAmount { get; set; }
    public bool HasExternalManagerWithholdingTaxExemption { get; set; }
    public string? ExternalManagerNationalId { get; set; }

    // -----------------------------------------------------------------
    // Tax configuration (applies to external company invoice)
    // HasVatExemption        = true  → skip the ×100/114 split; pay gross as-is
    // HasWithholdingTaxExemption = true  → do not deduct withholding from net
    // -----------------------------------------------------------------
    public bool HasVatExemption { get; set; }
    public bool HasWithholdingTaxExemption { get; set; }
    public decimal VatPercent { get; set; }
    public decimal WithholdingTaxPercent { get; set; }

    public string? Notes { get; set; }

    public DateTime? LastUpdatedStamp { get; set; }
    public DateTime? CreatedStamp { get; set; }

    // -----------------------------------------------------------------
    // Navigation
    // -----------------------------------------------------------------
    public virtual SalesRequest SalesRequest { get; set; } = null!;
    public virtual WorkEffort? Project { get; set; }
    public virtual Party? SalesRepParty { get; set; }
    public virtual Party? ManagerParty { get; set; }
    public virtual Party? SalesRep2Party { get; set; }
    public virtual Party? Manager2Party { get; set; }
    public virtual Party? ExternalCompanyParty { get; set; }
    public virtual Party? ExternalSalesRepParty { get; set; }
    public virtual Party? ExternalManagerParty { get; set; }
}
